bigquery/google/cloud/bigquery/job.py

```diff
@@ -2516,37 +2565,74 @@ class QueryPlanEntry(object):
     :type status: str
     :param status: entry status
 
+    :type shuffle_output_bytes: int
+    :param shuffle_output_bytes: number of bytes written to shuffle
+
+    :type shuffle_output_bytes_spilled: int
+    :param shuffle_output_bytes_spilled: number of bytes written
+            to shuffle and spilled to disk
+
     :type steps: List(QueryPlanEntryStep)
     :param steps: steps in the entry
     """
     def __init__(self,
                  name,
                  entry_id,
+                 start_ms,
```

Comment:
Rather than pass in a billion things to the constructor, how about we update this to take a resource object, which we save to `_properties` like our other classes?